test/envtest/konnect_entities_konnectdataplanegroupconfiguration_test.go

```diff
@@ -0,0 +1,120 @@
+			deploy.KonnectGatewayControlPlaneTypeWithCloudGatewaysEnabled(),
+		)
+
+		const (
+			id = "12345"
+		)
+
+		t.Log("Setting up a watch for KonnectCloudGatewayDataPlaneGroupConfiguration events")
+		w := setupWatch[konnectv1alpha1.KonnectCloudGatewayDataPlaneGroupConfigurationList](t, ctx, cl, client.InNamespace(ns.Name))
+		t.Log("Setting up SDK expectations on creation")
+		sdk.CloudGatewaysSDK.EXPECT().CreateConfiguration(
+			mock.Anything,
+			mock.MatchedBy(func(req sdkkonnectcomp.CreateConfigurationRequest) bool {
+				return req.ControlPlaneID == cp.GetKonnectID()
+			}),
+		).Return(&sdkkonnectops.CreateConfigurationResponse{
+			ConfigurationManifest: &sdkkonnectcomp.ConfigurationManifest{
+				ID:             id,
+				ControlPlaneID: cp.GetKonnectID(),
+			},
+		}, nil)
+
+		t.Log("Creating a KonnectCloudGatewayDataPlaneGroupConfiguration")
+		dpgconf := deploy.KonnectCloudGatewayDataPlaneGroupConfiguration(t, ctx, clientNamespaced,
+			deploy.WithKonnectNamespacedRefControlPlaneRef(cp),
+		)
+
+		t.Log("Waiting for KonnectCloudGatewayDataPlaneGroupConfiguration to be programmed and get Konnect ID")
+		watchFor(t, ctx, w, watch.Modified, func(c *konnectv1alpha1.KonnectCloudGatewayDataPlaneGroupConfiguration) bool {
+			return c.GetKonnectID() == id && k8sutils.IsProgrammed(c)
+		}, "KonnectCloudGatewayDataPlaneGroupConfiguration didn't get Programmed status condition or didn't get the correct (12345) Konnect ID assigned")
+
+		t.Log("Checking SDK operations")
+		require.EventuallyWithT(t, func(c *assert.CollectT) {
+			assert.True(c, factory.SDK.CloudGatewaysSDK.AssertExpectations(t))
+		}, waitTime, tickTime)
+
+		// NOTE: we delete the data plane group configuration by "creating" (using PUT)
+		// because Cloud Gateways DataPlane Group connfiguration API doesn't support
+		// deletion of the configuration directly.
+		t.Log("Setting up SDK expectations on deletion")
+		sdk.CloudGatewaysSDK.EXPECT().CreateConfiguration(
+			mock.Anything,
+			mock.MatchedBy(func(req sdkkonnectcomp.CreateConfigurationRequest) bool {
+				return req.ControlPlaneID == cp.GetKonnectID() &&
+					len(req.DataplaneGroups) == 0
+			}),
+		).Return(&sdkkonnectops.CreateConfigurationResponse{
+			ConfigurationManifest: &sdkkonnectcomp.ConfigurationManifest{
+				ID:             id,
+				ControlPlaneID: cp.GetKonnectID(),
+			},
+		}, nil)
+
+		t.Log("Deleting")
+		require.NoError(t, clientNamespaced.Delete(ctx, dpgconf))
+
+		t.Log("Waiting for object to disappear")
+		assert.EventuallyWithT(t, func(c *assert.CollectT) {
+			err := clientNamespaced.Get(ctx, client.ObjectKeyFromObject(dpgconf), dpgconf)
+			assert.True(c, err != nil && k8serrors.IsNotFound(err))
+		}, waitTime, tickTime)
+
+		t.Log("Waiting for object to be deleted in the SDK")
+		assert.EventuallyWithT(t, func(c *assert.CollectT) {
+			assert.True(c, factory.SDK.CloudGatewaysSDK.AssertExpectations(t))
+		}, waitTime, tickTime)
```

Comment:
Why not `require`  everywhere instead of `assert`?

Reply:
Changed to `require.`